controllers/clusterversion_controller.go

```diff
@@ -436,3 +445,40 @@ func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
 
 	return nil
 }
+
+func (c *ClusterVersionReconciler) reconcileCSIAddonsSubscription() error {
```

Comment:
dont we need to watch for subscription as well?

Reply:
I am just fetching the subscription and changing the channel if it doesn't match ocs-client-operator as my intention here was to change the channel for csi-addons if it doesn't get upgraded

Reply:
We should watch for subscription if not already, otherwise how this code will be triggered automatically if there is a change in the subscription

Reply:
added watch for subscriptions, verifying the code changes